creme/tree/_nodes/arf_htr_nodes.py

```diff
@@ -0,0 +1,135 @@
+from creme.tree._attribute_observer import NominalAttributeRegressionObserver
+from creme.tree._attribute_observer import NumericAttributeRegressionObserver
+from creme.utils.skmultiflow_utils import check_random_state
+
+from .htr_nodes import LearningNodeMean
+from .htr_nodes import LearningNodeModel
+from .htr_nodes import LearningNodeAdaptive
+from .arf_htc_nodes import RandomActiveLeafClass
+
+
+class RandomActiveLeafRegressor(RandomActiveLeafClass):
+    @staticmethod
+    def new_nominal_attribute_observer():
+        return NominalAttributeRegressionObserver()
+
+    @staticmethod
+    def new_numeric_attribute_observer():
+        return NumericAttributeRegressionObserver()
+
+    def manage_memory(self, criterion, last_check_ratio, last_check_sdr, last_check_e):
+        """ Trigger Attribute Observers' memory management routines.
+
+        Currently, only `NumericAttributeRegressionObserver` has support to this feature.
+
+        Parameters
+        ----------
+            criterion: SplitCriterion
+                Split criterion
+            last_check_ratio: float
+                The ratio between the second best candidate's merit and the merit of the best
+                split candidate.
+            last_check_sdr: float
+                The best candidate's split merit.
+            last_check_e: float
+                Hoeffding bound value calculated in the last split attempt.
+        """
+        for obs in self.attribute_observers.values():
+            if isinstance(obs, NumericAttributeRegressionObserver):
+                obs.remove_bad_splits(criterion=criterion, last_check_ratio=last_check_ratio,
+                                      last_check_vr=last_check_sdr, last_check_e=last_check_e,
+                                      pre_split_dist=self._stats)
+
+
+class RandomActiveLearningNodeMean(LearningNodeMean, RandomActiveLeafRegressor):
+    """ Learning Node for regression tasks that always use the average target
```

Comment:
As a general note, you don't need to leading space in docstrings.

Reply:
I removed the extra space characters in the recent commits.